src/Plotting/plot_recipes.jl

```diff
@@ -198,6 +198,75 @@ julia> plot(Bs, 1e-2)  # faster but less accurate than the previous call
     end
 end
 
+# recipe for vector of singletons
+@recipe function plot_list(list::Vector{SN}) where {N, SN<:AbstractSingleton{N}}
+
+    label --> DEFAULT_LABEL
+    grid --> DEFAULT_GRID
+    if DEFAULT_ASPECT_RATIO != :none
+        aspect_ratio --> DEFAULT_ASPECT_RATIO
+    end
+    seriesalpha --> DEFAULT_ALPHA
+    seriescolor --> DEFAULT_COLOR
+    seriestype --> :scatter
+
+    _plot_singleton_list(list)
+end
+
+# plot recipe for the union of singletons
+@recipe function plot_list(X::UnionSetArray{N, SN}) where {N, SN<:AbstractSingleton{N}}
+
+    label --> DEFAULT_LABEL
+    grid --> DEFAULT_GRID
+    if DEFAULT_ASPECT_RATIO != :none
+        aspect_ratio --> DEFAULT_ASPECT_RATIO
+    end
+    seriesalpha --> DEFAULT_ALPHA
+    seriescolor --> DEFAULT_COLOR
+    seriestype --> :scatter
+
+    list = array(X)
+    _plot_singleton_list(list)
+end
+
+function _plot_singleton_list(list::Vector{SN}) where {N, SN<:AbstractSingleton{N}}
+    n = dim(first(list))
+    if n == 1
+        _plot_singleton_list_1D(list)
+    elseif n == 2
+        _plot_singleton_list_2D(list)
+    else
+        throw(ArgumentError("plotting a vector of singletons is only available for dimensions " *
+             "one or two, got dimension $n"))
+    end
+end
+
+function _plot_singleton_list_1D(list::Vector{SN}) where {N, SN<:AbstractSingleton{N}}
+    m = length(list)
+    x = Vector{N}()
+    y = Vector{N}()
+
+    for Xi in list
+        p = element(Xi)
+        push!(x, p[1])
+        push!(y, zero(N))
```

Comment:
```suggestion
```suggestion
    x = Vector{N}(undef, m)
    y = zeros(N, m)

    @inbounds for (i, Xi) in enumerate(list)
        p = element(Xi)
        x[i] = p[1]
```